Sources/EpoxyCollectionView/CollectionView/ReusableViews/CollectionViewCell.swift

```diff
@@ -120,22 +120,31 @@ public final class CollectionViewCell: UICollectionViewCell, ItemCellView {
     // On iOS 15, cells that return an unstable size can result in a layout recursion crash. The
     // root cause of these crashes is likely an ambiguous layout due to misconfigured constraints.
     // Unfortunately, finding each and every one of these ambiguous layouts can be challenging. This
-    // code works around this new behavior / crash by giving up on self-sizing after 5 attempts.
-    // Once a component has been sized differently 5 times, we assume that it has an ambiguous
-    // layout and we simply return the last computed size for the remainder of the component's life.
-    // We reset the count on cell reuse and when the cell is reconfigured in `CollectionView`'s
-    // `configure` function.
-    if CollectionViewConfiguration.shared.enableLayoutRecursionWorkaround {
-      if let previousComputedSize = previousComputedSize, numberOfNewComputedSizes >= 5 {
-        EpoxyLogger.shared.assertionFailure(
-          "Layout recursion detected. View: \(view?.description ?? "nil view"). Size: \(preferredAttributes.size).")
-        preferredAttributes.size = previousComputedSize
+    // code tries to detect these layout issues and logs a warning when one is detected: if a
+    // component size is unstable for at least 50 sizing attempts in less than 2 seconds, we log a
+    // warning.
+
+    let sizingAttemptTime = CACurrentMediaTime()
+    let wasFirstSizingAttemptRecent: Bool
+    if let firstSizingAttemptTime = firstSizingAttemptTime {
+      wasFirstSizingAttemptRecent = sizingAttemptTime - firstSizingAttemptTime < 2
```

Comment:
I feel like we could probably lower this to something like 0.1s so that it would be less likely to hit in the case of device rotations or resizing text fields, what do you think? Since a frame draw is 0.016s and cell layouts are typically much faster than frame draws I think we're safe to lower this quite a bit

Reply:
+1

Reply:
I based this number based on a video of a crash that I saw QA post. Hard to know exactly what the right number is. If I lower it below 0.5s, then I also need to lower the layout-attempt threshold from 50 to something like 20. Related reply: https://github.com/airbnb/epoxy-ios/pull/88#discussion_r818164377